boards/earlgrey-nexysvideo/src/main.rs

```diff
@@ -404,12 +411,57 @@ pub unsafe fn main() {
     });
     debug!("OpenTitan initialisation complete. Entering main loop");
 
-    let scheduler = components::sched::priority::PriorityComponent::new(board_kernel).finalize(());
-    board_kernel.kernel_loop(
-        &earlgrey_nexysvideo,
-        chip,
-        None::<&kernel::ipc::IPC<NUM_PROCS>>,
-        scheduler,
-        &main_loop_cap,
-    );
+    (board_kernel, earlgrey_nexysvideo, chip)
+}
+
+/// Main function.
+///
+/// This function is called from the arch crate after some very basic RISC-V
+/// setup and RAM initialization.
+#[no_mangle]
+pub unsafe fn main() {
+    #[cfg(test)]
+    test_main();
```

Comment:
Where does this come from and what code is actually running?? This is why cfgs/macros can be so hard for new users. I'm searching for "fn test_main" in the changed files and not finding anything. This normally always works in Rust because `fn` has to be the last keyword before the function name. But either I'm missing something or rust testing breaks that primitive.

Reply:
This comes from the `reexport_test_harness_main` at the top of the file. There are some more details here: https://os.phil-opp.com/testing/#custom-test-frameworks

It seems to be part of Rust's test infrastructure.